CONTRIBUTING.md

```diff
@@ -0,0 +1,25 @@
+# Contributors Guide
+
+ If you believe that you have found a bug, please search for an existing [issue](https://issues.apache.org/jira/browse/accumulo) to see if it has already been reported. If you would like to add a new feature to Accumulo, please send an email with your idea to the [user](mailto:user@accumulo.apache.org) mail list. If it's appropriate, then we will create a ticket and assign it to you.
+
+## Development
+
+- Ensure that your work targets the correct branch
+- Add / update unit and integration tests
+- Ensure that Accumulo builds cleanly using the command: `mvn clean verify`
+
+## Patch Submission
+
+- Before submission please squash your commits using a message that contains the JIRA issue number and a description of the changes.
```

Comment:
How about: "using a message that starts with the JIRA issue number". I could see the current text being misconstrued (e.g. `foo bar baz ACCUMULO-XXXX` or `XXXX foo bar baz`)

We're quite consistent with commit messages being `ACCUMULO-XXXX foo bar baz` (with more text after a line break)